packages/starknet-snap/src/getErc20TokenBalance.ts

```diff
@@ -42,6 +43,6 @@ export async function getErc20TokenBalance(params: ApiParams) {
     return resp.result[0];
   } catch (err) {
     logger.error(`Problem found: ${err}`);
-    throw err;
+    throw new InternalError(err);
```

Comment:
simplify throw err from the function method and catch it from index.ts and throw the InternalError from index

or  

apply to InternalError to  all files, instead of just 1

Reply:
may be we just need to modify the index.ts add a big try catch would do the work